### PR TITLE
Fix for items weighting below 1g

### PR DESCRIPTION
**Pull request by RDru**
#### Summary
```SUMMARY: Bugfixes "Fix for items weighting below 1g"```

#### Purpose of change
Items weighting below 1g were rounded to 0.

#### Describe the solution
Not rounding to grams.

#### Describe alternatives you've considered
none

#### Testing
Tested items below 1g, also items with "X kg" in json. Tested saving and loading.

#### Additional context
Also updated json_info that mg, g and kg can be used in "weight".

### CODE REVIEW

File: src/item.cpp [hunk 1]
```diff
@@ -3845,11 +3845,11 @@ units::mass item::weight( bool include_contents, bool integral ) const
     }
 
     units::mass ret;
-    if( integral ) {
-        ret = units::from_gram( get_var( "integral_weight", to_gram( type->integral_weight ) ) );
-    } else {
-        ret = units::from_gram( get_var( "weight", to_gram( type->weight ) ) );
-    }
+    std::string local_str_mass = integral ? get_var( "integral_weight" ) : get_var( "weight" );
+    if( strlen( local_str_mass.c_str() ) > 0 )
```
Comment (Qrox) [inline src/item.cpp]:
```suggestion
    if( !local_str_mass.empty() )
```

Comment (RDru) [reply src/item.cpp]: ok, but without !, flipped condition

File: src/item.cpp [hunk 2]
```diff
@@ -3845,10 +3846,11 @@ units::mass item::weight( bool include_contents, bool integral ) const
...
 
     units::mass ret;
-    if( integral ) {
-        ret = units::from_gram( get_var( "integral_weight", to_gram( type->integral_weight ) ) );
+    std::string local_str_mass = integral ? get_var( "integral_weight" ) : get_var( "weight" );
+    if( local_str_mass.empty() ) {
+        ret = integral ? type->integral_weight : type->weight;
     } else {
-        ret = units::from_gram( get_var( "weight", to_gram( type->weight ) ) );
+        ret = units::from_milligram( _atoi64( local_str_mass.c_str() ) );
```
Comment (Qrox) [inline src/item.cpp]: Actually, I think you should use `has_var()` to check if the variables exist and use `get_var( "...", 0 )` to get the variable as an integer, so you can avoid using `_atoi64` here, which is not a standard function.

Comment (RDru) [reply src/item.cpp]: going to use std::stoll

File: src/item.cpp (hunk 1 above)
Comment (AMurkin) [reply src/item.cpp]: Length > 0 means it's *not* empty.

**Pull request merged**